src/force/CabanaPD_Force_PMB.hpp

```diff
@@ -0,0 +1,387 @@
+    {
+        auto c = _model.c;
+        const auto vol = particles.sliceVolume();
+
+        auto energy_full =
+            KOKKOS_LAMBDA( const int i, const int j, double& Phi )
+        {
+            // Get the bond distance, displacement, and stretch.
+            double xi, r, s;
+            getDistance( x, u, i, j, xi, r, s );
+
+            // 0.25 factor is due to 1/2 from outside the integral and 1/2 from
+            // the integrand (pairwise potential).
+            double w = 0.25 * c * s * s * xi * vol( j );
+            W( i ) += w;
+            Phi += w * vol( i );
+        };
+
+        double strain_energy = 0.0;
+        Kokkos::RangePolicy<exec_space> policy( 0, n_local );
+        Cabana::neighbor_parallel_reduce(
+            policy, energy_full, neigh_list, Cabana::FirstNeighborsTag(),
+            neigh_op_tag, strain_energy,
+            "CabanaPD::ForcePMB::computeEnergyFull" );
+
+        return strain_energy;
+    }
+};
+
+template <class ExecutionSpace>
+class Force<ExecutionSpace, ForceModel<PMB, Fracture>>
+    : public Force<ExecutionSpace, ForceModel<PMB, Elastic>>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, ForceModel<PMB, Elastic>>;
+    using base_type::_half_neigh;
+    ForceModel<PMB, Fracture> _model;
+
+  public:
+    using exec_space = ExecutionSpace;
+
+    Force( const bool half_neigh, const ForceModel<PMB, Fracture> model )
+        : base_type( half_neigh, model )
+        , _model( model )
+    {
+    }
+
+    template <class ForceType, class PosType, class ParticleType,
+              class NeighListType, class MuView, class ParallelType>
+    void computeForceFull( ForceType& f, const PosType& x, const PosType& u,
+                           const ParticleType& particles,
+                           const NeighListType& neigh_list, MuView& mu,
+                           const int n_local, ParallelType& ) const
+    {
+        auto c = _model.c;
+        auto break_coeff = _model.bond_break_coeff;
+        const auto vol = particles.sliceVolume();
+        const auto nofail = particles.sliceNoFail();
+
+        auto force_full = KOKKOS_LAMBDA( const int i )
+        {
+            std::size_t num_neighbors =
+                Cabana::NeighborList<NeighListType>::numNeighbor( neigh_list,
+                                                                  i );
+            for ( std::size_t n = 0; n < num_neighbors; n++ )
+            {
+                double fx_i = 0.0;
+                double fy_i = 0.0;
+                double fz_i = 0.0;
+
+                std::size_t j =
+                    Cabana::NeighborList<NeighListType>::getNeighbor(
+                        neigh_list, i, n );
+
+                // Get the reference positions and displacements.
+                double xi, r, s;
+                double rx, ry, rz;
+                getDistanceComponents( x, u, i, j, xi, r, s, rx, ry, rz );
+
+                // Break if beyond critical stretch unless in no-fail zone.
```

Comment:
@streeve Because we have ">=" we may say instead:
"// Break if reaches critical stretch unless in no-fail zone."

Reply:
Add it to your future "comments and clarification" PR